src/components/TodoList/TodoList.tsx

```diff
@@ -1 +1,117 @@
-export const TodoList = () => {};
+import { useState, useEffect } from 'react';
+import { TodoInfo } from '../TodoInfo';
+import { Todo } from '../../types/Todo';
+import usersFromServer from '../../api/users';
+import { getUser } from '../../utils/function';
+
+type Props = {
+  todos: Todo[],
+};
+
+function createNewTodo(
+  todos: Todo[],
+  id: number,
+  title: string,
+) {
+  return {
+    userId: id,
+    id: Math.max(...todos.map(todo => todo.id)) + 1,
+    title,
+    completed: false,
+    user: getUser(id, usersFromServer),
+  };
+}
+
+export const TodoList: React.FC<Props> = ({ todos }) => {
+  const [id, setId] = useState(0);
+  const [newTitle, setNewTitle] = useState('');
+  const [todoList, setTodoList] = useState(todos);
+
+  const [idError, setIdError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
```

Comment:
```suggestion
  const [isIdError, setIsIdError] = useState(false);
  const [isTitleError, setIsTitleError] = useState(false);
```
![image](https://github.com/mate-academy/react_add-todo-form/assets/96409563/db409d14-2ece-4c68-bea4-b42ab68c4128)